### DISCUSSION TITLE
Filtering tags from array

### DISCUSSION THREAD

**Discussion by Stanisla** [maplibre/martin#1405] [Q&A]
I'm trying to write my own style from OpenStreeMaps.
I have no idea how to filter an array like this in tags:
Object { highway: "motorway", oneway: "yes", osm_id: 24952578, ref: "AP-68", surface: "asphalt", **tags: '"toll"=>"yes", "lanes"=>"3", "int_ref"=>"E 804"**, ......"', z_order: 380, sourceLayer: "planet_osm_line" }
Any idea how can i filter "toll"=>"yes" vs "toll"=>"no" ?
Thanks in advance.

**Comment by HarelM** [comment]
This doesn't look like a valid json...?
Also, an array in properties when converted to mvt is not a great idea to parse...
I would suggest to create more properties in the tile generation process instead of relying on arrays.

**Reply by Stanisla** [reply]
This is the full response I get from map.queryRenderedFeatures()  :
` Object { highway: "motorway", oneway: "yes", osm_id: 24952578, ref: "AP-68", tags: '"toll"=>"yes", "lanes"=>"3", "int_ref"=>"E 804", "nat_ref"=>"A-68", "maxspeed"=>"120", "old_name"=>"Eusko-Aragoiar autobidea/Autopista Vasco-Aragonesa", "ref:colour"=>"#19408B", "old_name:es"=>"Autopista Vasco-Aragonesa", "old_name:eu"=>"Eusko-Aragoiar autobidea", "ref:colour_tx"=>"white"', z_order: 380, sourceLayer: "planet_osm_line" }`
The mvt is generated with martin from PostGis database feeded with an imported *.pbf
HarelM are you suggesting something like this   [https://maplibre.org/martin/sources-pg-functions.html](url) ? Any example apart from that?

**Reply by wipfli** [reply]
Probably best to ask in the Martin repo. Or @nyurik might be able to help you out...

**Reply by nyurik** [reply]
You are including `hstore` value `tags` in the output.  MVT does not handle dictionaries - so instead, PostGIS converts it into a string. That string is not understood by the MapLibre GL JS, as expected.  If you want to use that data, you have to create explicit columns for all that data.  My recommendation is first to use something like Planetiler to get everything up and running, and then perhaps adjust Planetiler's code to include more data types that you need. You do NOT want to include everything OSM has - as a lot of it would be irrelevant for a visual map representation.  You can then use Martin to serve tiles from the .mbtiles (from local disk) or .pmtiles (e.g. from a bucket).

**Reply by nyurik** [reply]
P.S. If you want to use SQL and construct tiles on the fly, you will probably want to use [function source](https://maplibre.org/martin/sources-pg-functions.html) -- where you would construct tiles with all the layers and needed columns using SQL.

**Reply by Stanisla** [reply]
Thanks nyurik.
I think i'll try the function source way.
If i get some results i'll post back.
